### PR TITLE
[Issue 6056][docs] Add example on how to choose partition when using a key

### LINKED ISSUE #6056: [docs] Add info on how to choose partition when using a key
**Describe the bug**
On [Partitioned topics](https://pulsar.apache.org/docs/en/next/cookbooks-partitioned/) page, it does not say what happens when you use a key. Does that override things? Does it change how a partition is chosen? Looking at the API docs, all of them talk about the absence of keys but don't say what happens with keys.

I see the link now to the routing modes. It doesn't really make sense that you can set round robin and have that overridden by simply adding a key.

With keys, how are partitions chosen? With Kafka it is:
selectedPartition = murmur2(keybytes) % numPartitions

Comments from Jesse Anderson

### PR DESCRIPTION
**Pull request by Jennifer88huang**
Fixes #6056 

### Motivation
There is no doc telling users how to choose partition when you use a key.

### Modifications
Add example on how to choose partition when using a key.

### CODE REVIEW

**Comment by Jennifer88huang** [comment]
run integration tests

**Comment by Jennifer88huang** [comment]
@zymap could you help review? Thank you.

**Comment by Jennifer88huang** [comment]
run integration tests
run java8 tests

**Comment by Jennifer88huang** [comment]
run integration tests
run java8 tests

**Comment by Jennifer88huang** [comment]
run java8 tests

**Comment by Jennifer88huang** [comment]
run java8 tests

**Comment by sijie** [comment]
run java8 tests

**Comment by Jennifer88huang** [comment]
run java8 tests

File: site2/docs/cookbooks-partitioned.md [hunk 1]
```diff
@@ -4,27 +4,25 @@ title: Partitioned topics
...
-You can [publish](#publishing-to-partitioned-topics) to partitioned topics using Pulsar's client libraries and you can [create and manage](#managing-partitioned-topics) partitioned topics using Pulsar's [admin API](admin-api-overview.md).
+You can publish to partitioned topics using Pulsar client libraries and you can [create and manage](#managing-partitioned-topics) partitioned topics using Pulsar [admin API](admin-api-overview.md).
 
-## Publishing to partitioned topics
+## Publish to partitioned topics
 
-When publishing to partitioned topics, the only difference from non-partitioned topics is that you need to specify a [routing mode](concepts-messaging.md#routing-modes) when you create a new [producer](reference-terminology.md#producer). Examples for [Java](#java) are below.
-
-### Java
+When publishing to partitioned topics, the only difference from non-partitioned topics is that you need to specify a [routing mode](concepts-messaging.md#routing-modes) when you create a new [producer](reference-terminology.md#producer). Take [Java](#java) as an example.
```
Comment (codelipenghui) [inline site2/docs/cookbooks-partitioned.md]: User does not have to explicitly specify a routing mode. If the user does not specify, RoundRobin route mode will be used.

File: site2/docs/cookbooks-partitioned.md [hunk 2]
```diff
@@ -72,6 +70,23 @@ Producer<byte[]> producer = pulsarClient.newProducer()
...
 
-## Managing partitioned topics
+### How to choose partitions when using a key
+If a message has a key, it supersedes the round robin routing policy. The following example illustrates how to choose partition when you use a key.
 
-You can use Pulsar's [admin API](admin-api-overview.md) to create and manage [partitioned topics](admin-api-partitioned-topics.md).
+```java
+// If the message has a key, it supersedes the round robin routing policy
+        if (msg.hasKey()) {
+            return signSafeMod(hash.makeHash(msg.getKey()), topicMetadata.numPartitions());
```
Comment (codelipenghui) [inline site2/docs/cookbooks-partitioned.md]:
```suggestion
            return signSafeMod(Murmur3_32Hash.getInstance().makeHash(msg.getKey()), topicMetadata.numPartitions());
```

File: site2/docs/cookbooks-partitioned.md (hunk 1 above)
Comment (Jennifer88huang) [reply site2/docs/cookbooks-partitioned.md]: Thank you very much for your advice. Updated, PTAL, thank you.

**Comment by codelipenghui** [comment]
retest this please

**Comment by Jennifer88huang** [comment]
retest this please

**Comment by sijie** [comment]
run integration tests

**Comment by codelipenghui** [comment]
run java8 tests

**Pull request merged**